Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -928,6 +930,40 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
     return 1;
 }
 
+int CLuaEngineDefs::EngineGetModelTextures(lua_State* luaVM)
+{
+    //  table engineGetModelTextures ( string modelName )
+    SString strModelName;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadString(strModelName, "");
+
+    if (argStream.HasErrors())
+        return luaL_error(luaVM, argStream.GetFullErrorMessage());
+
+    ushort usModelID = CModelNames::ResolveModelID(strModelName);
+    if (usModelID != INVALID_MODEL_ID)
+    {
+        std::vector<std::tuple<std::string, CPixels>> textureList;
+        g_pGame->GetRenderWare()->GetModelTextures(textureList, usModelID);
+
+        lua_newtable(luaVM);
+        for (auto pair : textureList)
+        {
+            CClientTexture* pTexture = g_pClientGame->GetManager()->GetRenderElementManager()->CreateTexture("", &std::get<1>(pair), RDEFAULT, RDEFAULT,RDEFAULT, RFORMAT_UNKNOWN, TADDRESS_WRAP);
+            lua_pushstring(luaVM, (const char*)&std::get<0>(pair));
```

Comment:
This does not look correct. Just do `std::get<0>(pair).c_str()` here instead of weird casts.